pkg/controller/utils/capability_test.go

```diff
@@ -386,3 +395,143 @@ variable "bbb" {
 		})
 	}
 }
+
+func TestGetGitSSHPublicKey(t *testing.T) {
+	sshAuth := make(map[string][]byte)
+	sshAuth[corev1.SSHAuthPrivateKey] = testdata.PEMBytes["rsa"]
+	sshAuth[GitCredsKnownHosts] = []byte(`github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa
+	+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7
+	VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKr
+	TJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==`)
+
+	pubKey, err := gitssh.NewPublicKeys("git", sshAuth[corev1.SSHAuthPrivateKey], "")
+	assert.NilError(t, err)
+
+	k8sClient := fake.NewClientBuilder().Build()
+	ctx := context.Background()
+
+	secret := corev1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "git-ssh-auth",
+			Namespace: "default",
+		},
+		Data: sshAuth,
+		Type: corev1.SecretTypeSSHAuth,
+	}
+	err = k8sClient.Create(ctx, &secret)
+	assert.NilError(t, err)
+
+	secret = corev1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "git-ssh-auth-no-ssh-privatekey",
+			Namespace: "default",
+		},
+		Data: map[string][]byte{
+			GitCredsKnownHosts: sshAuth[GitCredsKnownHosts],
+		},
+	}
+	err = k8sClient.Create(ctx, &secret)
+	assert.NilError(t, err)
+
+	secret = corev1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "git-ssh-auth-no-known_hosts",
+			Namespace: "default",
+		},
+		Data: map[string][]byte{
+			corev1.SSHAuthPrivateKey: sshAuth[corev1.SSHAuthPrivateKey],
+		},
+		Type: corev1.SecretTypeSSHAuth,
+	}
+	err = k8sClient.Create(ctx, &secret)
+	assert.NilError(t, err)
+
+	type args struct {
+		k8sClient                     client.Client
+		GitCredentialsSecretReference *corev1.SecretReference
+	}
+	type want struct {
+		publicKey *gitssh.PublicKeys
+		err       string
+	}
+	cases := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "git credentials secret not found",
+			args: args{
+				k8sClient: k8sClient,
+				GitCredentialsSecretReference: &corev1.SecretReference{
+					Name:      "git-shh-auth",
```

Comment:
This case can be a little illegible, let's make the name more explicit, like `auth-secret-not-exist`